fossunited/foss_hackathon/doctype/foss_hackathon_join_team_request/foss_hackathon_join_team_request.py

```diff
@@ -60,12 +64,31 @@ def reject_other_requests(self):
         requests = frappe.get_all(
             JOIN_TEAM_REQUEST,
             filters={
-                "team": self.team,
+                "hackathon": self.hackathon,
                 "status": "Pending",
                 "reciever_email": self.reciever_email,
+                "name": ["!=", self.name],
             },
+            pluck="name",
         )
         for request in requests:
-            request_doc = frappe.get_doc(JOIN_TEAM_REQUEST, request.name)
+            request_doc = frappe.get_doc(JOIN_TEAM_REQUEST, request)
             request_doc.status = "Rejected"
-            request_doc.save()
+            request_doc.save(ignore_permissions=True)
+
+    def validate_sender_as_member(self):
+        if frappe.permissions.is_system_user(frappe.session.user):
+            return
+
+        participant_id = frappe.db.get_value(
+            HACKATHON_PARTICIPANT, {"user": self.requested_by}, ["name"]
+        )
+
+        is_team_member = frappe.db.exists(
+            HACKATHON_TEAM_MEMBER, {"parent": self.team, "member": participant_id}
+        )
+
+        if not is_team_member:
+            frappe.throw("Request sender is not a team member", frappe.ValidationError)
```

Comment:
i'm not sure how this can happen in a real-world scenario. how does someone send a join request to a user while not being a team member themselves? here again I'm wondering if this code is influenced by the scenario being tested or if it can really happen in production

Reply:
It can happen since anyone can create a Join Request, it is just a doctype. We have to make sure that it doesn't through these controller methods

Reply:
is this scenario possible? From the UI

- ansh, you, and I are registered for foss hack
- you create a team for foss hack
- I send ansh a request to join your team

Having never seen the UI. I'm surprised that there is a way to do this. could we not prevent this in the UI i.e. I can only invite someone to join my team - assuming I created a team in the first place.

Reply:
It can't be done via the ui, no. Only through api calls

Reply:
I was afraid so. can we please add this info to the docstring here as well?